pkg/serviceaccount/legacy.go

```diff
@@ -77,6 +80,10 @@ func (v *legacyValidator) Validate(tokenData string, public *jwt.Claims, private
 	if len(namespace) == 0 {
 		return nil, errors.New("namespace claim is missing")
 	}
+	tenant := private.Tenant
```

Comment:
I remember in current implementation  obj.tenant could be empty if it's in system space. Will this check break the service account functionality in system space?

Reply:
what this check does is to extract tenant from the token provided "user". subsequently this tenant is used to extract user's other info for validation. if the token provided by user does not contain a tenant info, I don't think we should default it to "system", otherwise any user service account without a tenant in their token would end up being authenticated as a service account of the "system" tenant.

Reply:
Thanks for clarification. At first I thought it was a resource object.